### PR TITLE
MAINT: Fix build failures and execution timeouts

### PR DESCRIPTION
**Pull request by rossbar**
This PR combines a plethora of tweaks and fixes to make the build/test/deploy infrastructure more reliable. The end result should be a reduction in CI failures due to things like resource requirements and execution timeouts.

Note this also includes some changes to the tutorials themselves, most notably the deep learning tutorials, to improve their usability in resource-constrained environments binder. Two changes to tutorial content need special attention:
 1. The evaluation step of the mnist tutorial has been vectorized. This has the advantage of demonstrating vectorization, but also significantly improves run-time performance.
 2. Reducing the number of episodes in the RL tutorial from 10 -> 3 (along with batch size). This is entirely a run-time argument: running the deep-rl tutorial in e.g. CI or binder would take on the order 10 minutes for the training step. Furthermore, since there is no tangible output from the training, I decided to reduce it further to make the tutorial more usable. @8bitmp3 your opinion on whether this is acceptable would be much appreciated.

Most of the other changes are related to minor build fixes (broken links, docs missing from toctrees, etc.) or additional CI infrastructure necessary for new dependencies (particularly `atari-py`). Once this PR is in we can also cache the tutorial dependencies, which will significantly reduce the build time as we can skip having to build the `atari-py` wheel from scratch every run. I'd also like to experiment with caching the tutorial execution during the build process, which depends on this as well.

### CODE REVIEW

**Comment by rossbar** [comment]
There's also an issue with the `atari-py` dependency - see #69. b7e8c3c is a temporary workaround for CI.

**Review by melissawm — COMMENTED** [review]
LGTM, @rossbar ! Thanks!

**Comment by 8bitmp3** [comment]
@rossbar that's some magic! Thank you 🙏 

> Reducing the number of episodes in the RL tutorial from 10 -> 3 (along with batch size). This is entirely a run-time argument: running the deep-rl tutorial in e.g. CI or binder would take on the order 10 minutes for the training step. Furthermore, since there is no tangible output from the training, I decided to reduce it further to make the tutorial more usable. @8bitmp3 your opinion on whether this is acceptable would be much appreciated.

The original 10 steps were too low but we may it clear here: _"...let's limit the number of episodes for training to 3. If you are using hardware acceleration (CPUs and GPUs), you can increase the number to 1,000 or beyond."_

> The evaluation step of the mnist tutorial has been vectorized. This has the advantage of demonstrating vectorization, but also significantly improves run-time performance.

That's amazing.
